src/main.rs

```diff
@@ -54,6 +64,42 @@ pub fn list_keys() -> Result<()> {
     Ok(())
 }
 
+pub fn listen(name: String, phys: Option<String>) -> Result<()> {
+    let device_info = deviceinfo::DeviceInfo::with_name(
+        &name,
+        phys.as_deref(),
+    )?;
+    let path = device_info.path.as_path();
+    let f = std::fs::File::open(path).context(format!("opening {}", path.display()))?;
+    let input_device = Device::new_from_file(f)
+        .with_context(|| format!("failed to create new Device from file {}", path.display()))?;
+    log::info!("Going into read loop");
+    loop {
+        let (status, event) = input_device.next_event(ReadFlag::NORMAL | ReadFlag::BLOCKING)?;
+        match status {
+            evdev_rs::ReadStatus::Success => {
+                match event.event_code {
+                    EventCode::EV_KEY(_) |
+                    EventCode::EV_REL(_) |
+                    EventCode::EV_ABS(_) |
+                    EventCode::EV_SW(_)  |
+                    EventCode::EV_LED(_) |
+                    EventCode::EV_SND(_) |
+                    EventCode::EV_REP(_) |
+                    EventCode::EV_FF(_)  |
+                    EventCode::EV_PWR    |
+                    EventCode::EV_FF_STATUS(_) |
+                    EventCode::EV_UNK { .. } |
+                    EventCode::EV_MAX =>
+                        log::info!("IN code: {:?} value: {:?}", event.event_code, event.value),
+                    _ => {}
```

Comment:
what are the other cases?
Can we eliminate the match and just unconditionally log the `event_code` and `value`?